### ISSUE TITLE
run OnkoDICOM

### ISSUE THREAD

**Issue by thanhtaiphys** [didymo/OnkoDICOM#248]
Hi

I followed the installation instruction at https://github.com/didymo/OnkoDICOM/wiki/Installation-Instructions and don't have any error for each step but the final step (python main.py) I can not run OnkoDICOM

![image](https://user-images.githubusercontent.com/16337715/200848479-941fc489-5eb5-4dea-9ddb-5f7af53b9999.png)

**Comment by sjswerdloff** [comment]
It's not clear which operating system you are using, but it looks like you need to install qt6 (which itself is not python or a python package)
if you are using Ubuntu 22.04:

sudo add-apt-repository universe
sudo apt update
sudo apt install qt6-base-dev

if you are using an older version of Ubuntu you might need:
sudo add-apt-repository "deb http://archive.ubuntu.com/ubuntu $(lsb_release -sc) universe"
sudo apt update
sudo apt install qt6-base-dev

**Comment by didymo** [comment]
Sorry, OnkoDICOM now uses poetry to manage dependences. THe Unbuntu instructions have been updated. The other instructions will be updated in due course. If you use Windows there is an executable available.

**Comment by thanhtaiphys** [comment]
I am running on Ubuntu 20.04

sudo add-apt-repository universe ==> done
sudo apt update ==> done
sudo apt install qt6-base-dev==> done

python main.py==> can not run

![image](https://user-images.githubusercontent.com/16337715/200944549-37d44dbc-965e-4c6a-94e8-166f44f982b5.png)

**Comment by sjswerdloff** [comment]
@thanhtaiphys , you can try installing locate
`sudo apt install plocate`
and then try to locate the library:
`locate libQt6Core.so.6`
and then (with whatever the actual path is)
`ldd /path/to/libQt6Core.so.6`
to see if there are any missing libraries that it calls on.  There may be some dependencies that are not being fulfilled.

I'll try to replicate on an Ubuntu 20.04 VM on Intel/AMD platform, but I probably can't get to that until this weekend.

**Comment by sjswerdloff** [comment]
I'm having some difficulty getting OnkoDICOM to run on macOS Monterey (12.5.1) on Apple Silicon (Arm64).
I had to (re?) install a number of packages:
```
pip install pyside6 
pip install dicompyler-core
pip install alphashape
pip install platipy
pip install imblearn
pip install kaplanmeier
pip install fuzzywuzzy
pip install pymedphys
pip install vtk
pip install python-Levenshtein

```

With that (actually, the python-Levenshtein isn't absolutely necessary, but there was a warning about performance without it), I got the welcome screen.

When I choose "Individual Patient Curation", it crashes.  I'll open a separate issue, but put the information here as well.
If I choose "Batch Processing" I get a follow-on dialog.

The crash has the following message:

PDPI: 127.50000191485792
qt.qpa.fonts: Populating font family aliases took 160 ms. Replace uses of missing font family "Segoe UI" with one that exists to avoid this cost. 
Multithreading with maximum 10 threads
<class 'TypeError'> int() argument must be a string, a bytes-like object or a real number, not 'ItemFlag' <traceback object at 0x2a64b1c40>
Traceback (most recent call last):
  File "/Users/stuartswerdloff/PythonProjects/OnkoDICOM/src/Controller/TopLevelController.py", line 82, in show_open_patient
    self.open_patient_window.check_selected_items()
  File "/Users/stuartswerdloff/PythonProjects/OnkoDICOM/src/View/OpenPatientWindow.py", line 429, in check_selected_items
    checked_nodes = self.get_checked_nodes(
  File "/Users/stuartswerdloff/PythonProjects/OnkoDICOM/src/View/OpenPatientWindow.py", line 578, in get_checked_nodes
    recurse(root)
  File "/Users/stuartswerdloff/PythonProjects/OnkoDICOM/src/View/OpenPatientWindow.py", line 571, in recurse
    if int(child.flags()) & int(Qt.ItemIsUserCheckable) and \
TypeError: int() argument must be a string, a bytes-like object or a real number, not 'ItemFlag'

**Comment by thanhtaiphys** [comment]
> 
I followed your comment

![image](https://user-images.githubusercontent.com/16337715/201424430-37fdf177-2a0c-4698-a147-4a8906e2b7d4.png)

**Comment by sjswerdloff** [comment]
please try 
`ldd /usr/lib/x86_64-linux-gnu/libQt6Core.so.6`
and
`ldd /home/td/OnkoDICOM/venv/lib/python3.8/site-packages/PySide6/Qt/lib/libQt6Core.so.6`

**Comment by didymo** [comment]
Sorry I have been using the Windows System for weeks to get a build. The fact that the Linux install became painful at that time was overlooked. The current list of pip installs Stuart provided will be investigated. Sorry for the misleading reply.

**Comment by sjswerdloff** [comment]
@thanhtaiphys   I managed to get OnkoDICOM 0.05.02-beta to start up on Ubuntu 20.04, but I had to do a fair amount of installation hacking that I don't think is reasonable to expect from  users.  Once I did get it running (first window), I had similar issues on Ubuntu 20.04 as I did on MacOS 12.5.1 Arm64 (Apple Silicon) having to do with forced recursive imports (of things that were deliberately intended to be imported only if needed).  And ultimately, there are packages that aren't needed for OnkoDICOM that get imported by force that are not compatible with python 3.10 and later.

So I'm of the opinion that OnkoDICOM 0.05.02-beta is MS Windows Only, and probably should only be used from the pre-built binary.
I'll work with the maintainer to fix the various things that are broken, and aim for something like a 0.05.03-beta version.
This will probably take several weeks.

**Comment by sjswerdloff** [comment]
@thanhtaiphys :  There are a few things to try,  I think the fundamental issue for the error message you get is a **missing dependency** (something that libQt6.Core.so.6 is looking for and is not finding).
Also, for Ubuntu, it seems one is **much better off with Python 3.9** (all kinds of problems go away)
 one should start with a virtual environment 
that uses python 3.9, not python 3.10:
(after installing pyenv)
The following is an example, I just ran it (but if your version of 3.9 is not 3.9.15, then specify whatever it is that you do get)
```
  pyenv install 3.9
  pyenv virtualenv 3.9.15 onko_py39
  pyenv activate onko_py39

``` 
please try installing these supporting (non-python) packages using apt:
```
sudo apt update
sudo apt-get -y install build-essential openssl libssl-dev libssl1.0 libgl1-mesa-dev libqt5x11extras5
sudo apt-get -y install libglu1-mesa-dev freeglut3-dev mesa-common-dev
sudo apt-get -y install g++

```
And then try using the python interpreter and a single line to see that PySide6 is actually fully available.
`python`
`>>> import PySide6`

if this fails, you might want to try with a fresh Ubuntu (and maybe start with Ubuntu 22.04... but repeat these instruction suggestions in that OS.  I got things to work on Ubuntu 22.04. first)
On the other hand, if this succeeds, You need to get poetry to update (and overwrite) the previous set of dependencies that were built on Windows with Python 3.10 in the OnkoDICOM directory:
```
poetry update
poetry install
```
I had to repeat the pip installation of pyradiomics (it failed on poetry install not finding numpy for some reason)
`pip install pyradiomics`

If you have crashes at that point, that aren't about PySide6, I can provide you with access to some "patches" (updated files), or I'll get some pull requests through the project (and you can pull them or re-clone).  I've submitted issues regarding the problems I ran in to (so you might want to take a look at those should you have further issues).

Sorry for the "teething pains"

**Comment by thanhtaiphys** [comment]
Thank you so much

**Comment by sjswerdloff** [comment]
@thanhtaiphys have  you gotten OnkoDICOM running yet?
I followed the instructions that are on the wiki starting from a clean python 3.9 virtual environment on Ubuntu 20.04 (but with all of the apt packages already installed), and it worked for me.
I'd like to close this issue, but I don't want to do that until you have OnkoDICOM running on your Ubuntu 20.04 environment

**Comment by thanhtaiphys** [comment]
Dear Sjswerdloff,

Thank you for your time to help me. 

Yes, please close this issue

Thanks

**Comment by sjswerdloff** [comment]
Closed at 80b9e7b0d5fc925d392500588f03d508a8705532
Ubuntu 20.04, python 3.9, apt packages to include (@didymo please update installation instructions to ensure the following are available in 20.04 and are in the list):
build-essential openssl libssl-dev libssl1.0 libgl1-mesa-dev libqt5x11extras5 libglu1-mesa-dev freeglut3-dev 
mesa-common-dev g++

**Comment by sjswerdloff** [comment]
@thanhtaiphys thank you for sticking with us while we work through some of the rough edges of the new milestone (0.5)
